php-classes/Slate/CBL/Competency.php

```diff
@@ -136,6 +136,16 @@ public function getSkillIds($forceRefresh = false)
         return $skillIds;
     }
 
+    private $totalSkills;
+    public function getTotalSkills($forceRefresh = false)
+    {
+        if (!empty($this->totalSkills)) {
+            return $this->totalSkills;
+        }
+
+        return $this->totalSkills = count($this->getSkillIds());
+    }
```

Comment:
@nbey better pattern is to frame these functions like:

```php
    public function getTotalSkills($forceRefresh = false)
    {
        if ($this->totalSkills === null) {
            $this->totalSkills = count($this->getSkillIds());
        }

        return $this->totalSkills;
    }
```

(`=== null` is needed because `0` is a legit value that would pass `== null`)